python-package/lightgbm/dask.py

```diff
@@ -296,11 +312,18 @@ def _train(
     # find an open port on each worker. note that multiple workers can run
     # on the same machine, so this needs to ensure that each one gets its
     # own port
-    worker_address_to_port = _find_ports_for_workers(
-        client=client,
-        worker_addresses=worker_map.keys(),
-        local_listen_port=params["local_listen_port"]
-    )
+    if params["local_listen_port"] == 12400:
+        # If local_listen_port is set to LightGBM's default value
+        # then we find a random open port for each worker
+        worker_address_to_port = client.run(_find_random_open_port)
```

Comment:
Can you please add `workers=worker_map.keys()` to this call? That would avoid running this on workers that don't hold any of the training data.